packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+        ownersContent: 'Owners Content',
+      };
+      const createResultSetName = `create${modelName}`;
+      const updateResultSetName = `update${modelName}`;
+      const privateResultSet = `
+        id
+        privateContent
+      `;
+      const ownerResultSet = `
+        ${privateResultSet}
+        owner
+        authors
+      `;
+      const setWithOwnerContent = `
+        ${ownerResultSet}
+        ownerContent
+      `;
+      const setWithOwnersContent = `
+        ${ownerResultSet}
+        ownersContent
+      `;
+      const completeOwnerResultSet = `
+        ${ownerResultSet}
+        ownerContent
+        ownersContent
+      `;
+
+      // owner(user1) creates a record with only allowed fields
+      const createResult1 = await user1TodoHelper.create(createResultSetName, todo, setWithOwnersContent);
+      expect(createResult1.data[createResultSetName].id).toBeDefined();
+      expect(createResult1.data[createResultSetName].owner).toEqual(userName1);
+      expect(createResult1.data[createResultSetName].authors).toEqual([userName1]);
+      todo['id'] = createResult1.data[createResultSetName].id;
+      todo['owner'] = userName1;
+      // protected fields are nullified in mutation responses
+      expectNullFields(createResult1.data[createResultSetName], ['privateContent', 'ownersContent']);
+
+      // user1 can update the allowed fields and add user2 to dyamic owners list field
+      const todoUpdated1 = {
+        id: todo['id'],
+        authors: [userName1, userName2],
+        privateContent: 'Private Content updated',
+        ownerContent: 'Owner Content',
+      };
+      const updateResult1 = await user1TodoHelper.update(updateResultSetName, todoUpdated1, setWithOwnerContent);
+      expect(updateResult1.data[updateResultSetName].id).toEqual(todo['id']);
+      expect(updateResult1.data[updateResultSetName].owner).toEqual(userName1);
+      expect(updateResult1.data[updateResultSetName].authors).toEqual([userName1, userName2]);
+      expectNullFields(updateResult1.data[updateResultSetName], ['privateContent', 'ownerContent']);
+
+      // user1 can read the allowed fields
+      const getResult1 = await user1TodoHelper.get(
+        {
+          id: todo['id'],
+        },
+        completeOwnerResultSet,
+        false,
+      );
+      checkOperationResult(getResult1, { ...todo, ...todoUpdated1 }, `get${modelName}`);
+
+      const listTodosResult1 = await user1TodoHelper.list({}, completeOwnerResultSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['id'], true);
+
+      // unless one has delete access to all fields in the model, delete is expected to fail
+      await expect(
+        async () => await user1TodoHelper.delete(`delete${modelName}`, { id: todo['id'] }, privateResultSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(`delete${modelName}`, 'Mutation'));
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket